tests/unit/mixins/onThisPageRegistrator.spec.js

```diff
@@ -0,0 +1,153 @@
+    }, {
+      kind: SectionKind.properties,
+      title: 'Properties Title',
+    }, {
+      kind: SectionKind.restBody,
+      title: 'restBody Title',
+    }, {
+      kind: SectionKind.restCookies,
+      title: 'restCookies Title',
+    }, {
+      kind: SectionKind.restEndpoint,
+      title: 'restEndpoint Title',
+    }, {
+      kind: SectionKind.restHeaders,
+      title: 'restHeaders Title',
+    }, {
+      kind: SectionKind.restParameters,
+      title: 'restParameters Title',
+    }, {
+      kind: SectionKind.restResponses,
+      title: 'restResponses Title',
+    }, {
+      kind: SectionKind.declarations,
+      content: ['foo'],
+    }, {
+      kind: SectionKind.details,
+      content: ['foo'],
+    }, {
+      kind: SectionKind.parameters,
+      content: ['foo'],
+    }, {
+      kind: SectionKind.possibleValues,
+      content: ['foo'],
+    },
+  ],
+  topicSections: [],
+  defaultImplementations: [],
+  relationshipsSections: [],
+  seeAlsoSections: [],
+});
+
+const createWrapper = attrs => shallowMount({
+  template: '<div/>',
+  mixins: [onThisPageRegistrator],
+  data: () => ({ topicData: topicData(), store: onThisPageSectionsStoreBase }),
+}, attrs);
+
+describe('OnThisPageRegistrator', () => {
+  beforeEach(() => {
+    onThisPageSectionsStoreBase.resetPageSections();
+  });
+
+  it('extracts the sections from the JSON', () => {
+    createWrapper();
+    expect(onThisPageSectionsStoreBase.state.onThisPageSections).toMatchSnapshot();
+  });
+
+  it('only stores headings up to level 3, anchorising `title` if no `anchor` is provided', async () => {
+    // clear our initial data
+    const wrapper = createWrapper({
+      data: () => ({ topicData: null }),
+    });
+    await flushPromises();
+    wrapper.setData({
+      topicData: {
+        primaryContentSections: [
+          {
+            kind: SectionKind.content,
+            content: [
+              ...contentSections,
+              {
+                type: 'heading',
+                level: 5,
+                text: 'Heading Level 5',
+              },
+            ],
+          },
+        ],
+      },
+    });
+    await flushPromises();
+    expect(onThisPageSectionsStoreBase.state.onThisPageSections).toEqual([
+      {
+        anchor: 'provided-heading-anchor',
+        level: 2,
+        title: 'Heading Level 2',
+      },
+      {
+        anchor: 'heading-level-3',
+        level: 3,
+        title: 'Heading Level 3',
+      },
+    ]);
+  });
+
+  it('watches for changes, clears the store and extracts sections again', async () => {
+    const wrapper = createWrapper();
+    expect(onThisPageSectionsStoreBase.state.onThisPageSections).toHaveLength(16);
+    wrapper.vm.topicData = {
+      primaryContentSections: [
+        {
+          kind: SectionKind.content,
+          content: contentSections,
+        },
+      ],
+    };
+    await flushPromises();
+    expect(onThisPageSectionsStoreBase.state.onThisPageSections).toHaveLength(2);
```

Comment:
Should we also do a check to make sure that it's the right two sections after things are updated? (Maybe overkill, feel free to ignore)